tests/e2e/cmp_test.go

```diff
@@ -379,13 +379,20 @@ var _ = Describe("odoCmpE2e", func() {
 })
 
 // ensures that the DeploymentConfig of the specified component
-// has completely rolled out
+// has completely rolled out and that none of the old pods are running
 // this is very useful to avoid race conditions that can occur when
 // updating the component
-func waitForDCOfComponentToRollout(componentName string) {
-	dcName := runCmd(fmt.Sprintf("oc get dc -l app.kubernetes.io/component-name=%s -o name", componentName))
+func waitForDCOfComponentToRolloutCompletely(componentName string) {
+	fullDCName := runCmd(fmt.Sprintf("oc get dc -l app.kubernetes.io/component-name=%s -o name | tr -d '\n'", componentName))
 	// oc rollout status ensures that the existing DC is fully rolled out before it terminates
 	// we need this because a rolling DC could cause odo update to fail due to its use
 	// of the read/update-in-memory/write-changes pattern
-	runCmd("oc rollout status " + dcName)
+	runCmd("oc rollout status " + fullDCName)
+
+	simpleDCName := strings.Replace(fullDCName, "deploymentconfig.apps.openshift.io/", "", -1)
+	// ensure that no more changes will occur to the name DC by waiting until there is only one pod running (the old one has terminated)
+	waitForEqualCmd(fmt.Sprintf("oc get pod -o name -l deploymentconfig=%s | wc -l | tr -d '\n'", simpleDCName), "1", 2)
+
+	// done in order to make sure that Openshift has updated the DC with the latest events
+	time.Sleep(5 * time.Second)
```

Comment:
I don't like `time.Sleep` being added, but this is a necessary evil to get the tests to not flake as much.

This LGTM.

Reply:
Unfortunately this PR still does not solve the problem.

It's still showing up sometimes... I was even able to reproduce it once on my own (not in a script) but so far I have not been able to pinpoint the problem nor of course come up with a proper fix